caddyhttp/staticfiles/fileserver.go

```diff
@@ -201,6 +201,7 @@ func (fs FileServer) serveFile(w http.ResponseWriter, r *http.Request, name stri
 
 		w.Header().Add("Vary", "Accept-Encoding")
 		w.Header().Set("Content-Encoding", encoding)
+		w.Header().Set("Content-Length", strconv.FormatInt(encodedFileInfo.Size(), 10))
```

Comment:
Hmm interesting question. ServeContent() doesn't set Content-Length when Content-Encoding is set.

```go
if w.Header().Get("Content-Encoding") == "" {
	w.Header().Set("Content-Length", strconv.FormatInt(sendSize, 10))
}
```

I opened an issue in the go project: https://github.com/golang/go/issues/19420